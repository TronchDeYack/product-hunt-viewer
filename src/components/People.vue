<template>
  <div class="people">
    <h2>Hunter</h2>
    <User :name="user.name" :img="user.profileImage" :headline="user.headline" />
    <h2>Makers</h2>
    <User v-for="maker in makers" :key="maker.username"
      :name="maker.name" :img="maker.profileImage" :headline="maker.headline" />
  </div>
</template>

<style lang="scss" scoped>
.people {
  background: white;
  border-radius: 5px;
  display: flex;
  flex-direction: column;

  h2 {
    font-size: 15px;
    text-align: left;
    margin-left: 15px;
  }
}
</style>

<script>
import User from '@/components/User.vue';

export default {
  name: 'authors',
  components: {
    User,
  },
  props: ['user', 'makers'],
};
</script>
