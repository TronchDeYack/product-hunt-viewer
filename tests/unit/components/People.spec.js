import { shallowMount } from '@vue/test-utils';

import People from '@/components/People.vue';

describe('People', () => {
  it('Should render.', () => {
    const wrapper = shallowMount(People, {
      propsData: {
        user: {
          name: 'Quos T.',
          username: 'quost',
          profileImage: 'https://ph-avatars.imgix.net/2250227/original?auto=format&fit=crop&crop=faces&w=original&h=original',
          headline: 'DevOps Engineer',
        },
        makers: [
          {
            name: 'Barbara Bermes',
            username: 'bbinto',
            headline: 'Sr. Product Manager, Firefox Android',
            profileImage: 'https://ph-avatars.imgix.net/2272615/original?auto=format&fit=crop&crop=faces&w=original&h=original',
          },
          {
            name: 'Anthony Lam',
            username: 'antlam',
            headline: 'Product Design Manager, Firefox UX',
            profileImage: 'https://ph-avatars.imgix.net/2272615/original?auto=format&fit=crop&crop=faces&w=original&h=original',
          },
          {
            name: 'Nick Nguyen',
            username: 'osunick',
            headline: 'Vice President, Firefox Product',
            profileImage: 'https://ph-avatars.imgix.net/2272615/original?auto=format&fit=crop&crop=faces&w=original&h=original',
          },
        ],
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
