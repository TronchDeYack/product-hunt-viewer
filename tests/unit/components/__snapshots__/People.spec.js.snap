// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People Should render. 1`] = `
<div class="people">
  <h2>Hunter</h2>
  <user-stub name="Quos T." headline="DevOps Engineer" img="https://ph-avatars.imgix.net/2250227/original?auto=format&amp;fit=crop&amp;crop=faces&amp;w=original&amp;h=original"></user-stub>
  <h2>Makers</h2>
  <user-stub name="Barbara Bermes" headline="Sr. Product Manager, Firefox Android" img="https://ph-avatars.imgix.net/2272615/original?auto=format&amp;fit=crop&amp;crop=faces&amp;w=original&amp;h=original"></user-stub>
  <user-stub name="Anthony Lam" headline="Product Design Manager, Firefox UX" img="https://ph-avatars.imgix.net/2272615/original?auto=format&amp;fit=crop&amp;crop=faces&amp;w=original&amp;h=original"></user-stub>
  <user-stub name="Nick Nguyen" headline="Vice President, Firefox Product" img="https://ph-avatars.imgix.net/2272615/original?auto=format&amp;fit=crop&amp;crop=faces&amp;w=original&amp;h=original"></user-stub>
</div>
`;
